src/django_svcs/middleware.py

```diff
@@ -1,16 +1,39 @@
-import svcs
+import asyncio
+from typing import Callable
+
 from django.http import HttpRequest, HttpResponse
+from django.utils.decorators import sync_and_async_middleware
+
+from .apps import svcs_from
 
-from . import apps
+# pyright: reportRedeclaration=false
```

Comment:
I'm not sure why this would be necessary.

Reply:
It fails without this, because pyright doesn't like the fact that both code paths there define a function named `middleware` but with different signatures (since one is async and one sync).